src/cockpit/jsonutil.py

```diff
@@ -77,6 +77,10 @@ def get_str(obj: JsonObject, key: str, default: Union[DT, _Empty] = _empty) -> U
     return _get(obj, lambda v: typechecked(v, str), key, default)
 
 
+def get_str_or_none(obj: JsonObject, key: str, default: Optional[str]) -> Optional[str]:
```

Comment:
This is a nice new helper. The commit message talks about "a couple of places", this commit could replace them, so that this isn't just dead code?

Reply:
FTR, it's not dead code as the next commit uses it, and this could also become a follow-up

Reply:
I also plan to use this as a stop-gap for fixing the kill message problem (where we accidentally send None for host/group).  Then we could add type safety while still allowing None as a value.  That's a separate PR, though.